UserInterface.cpp

```diff
@@ -665,7 +665,7 @@ std::optional<LaunchOptions> runControlCenter(const std::wstring& ra3Path, const
 		auto childDC = reinterpret_cast<HDC>(hdcAddress);
 		setBackgroundMode(childDC, TRANSPARENT).release();
 		setTextColor(childDC, RGB(255, 255, 255)).release();
-		return reinterpret_cast<INT_PTR>(commandLineBackground.get()); //per https://msdn.microsoft.com/en-us/library/windows/desktop/bb761691(v=vs.85).aspx
+		return (int)reinterpret_cast<INT_PTR>(commandLineBackground.get()); //per https://msdn.microsoft.com/en-us/library/windows/desktop/bb761691(v=vs.85).aspx

```

Comment:
If you are getting a compiler error on this line, it's probably because you are compiling the project as a x64 executable: https://github.com/lanyizi/RA3Bar-RA3Launcher/issues/3
In 32-bit mode, the `INT_PTR` is same as `int` (they're both 32 bits) so there won't be any issues, but in 64-bit mode, `INT_PTR` has 64 bits so you get the compile error.

If you want to compile it under 64-bit mode, then you **shouldn't** cast the return value to `int`, otherwise you will truncate the 64-bit handle to a 32-bit integer.
Instead, you could explicitly define the function return type on this line:
https://github.com/lanyizi/RA3Bar-RA3Launcher/blob/68c03bd2acc6be937615a1e8d7d0e5af0e698f40/UserInterface.cpp#L660
By writing 
_handlers[WM_CTLCOLOREDIT] = [&commandLineBackground](HWND window, WPARAM hdcAddress, LPARAM childAddress) **-> INT_PTR** {_
<br/>
PS: Actually this issue is addressed in [another pull request](https://github.com/lanyizi/RA3Bar-RA3Launcher/pull/6) too:
https://github.com/lanyizi/RA3Bar-RA3Launcher/blob/b8a522062a13d861f9a0373a5b0a12cf19df9d6c/UserInterface.cpp#L658
But unfortunately looks like the author abandoned his pull request...

Reply:
OK, I will edit github actions and make it complies launcher as 32 executable.
For a launcher, 32-bit seems enough, and for compatibility.